### PR TITLE
Tests: migrate JavaScript tests from karma to jasmine framework

### LINKED ISSUE #12141: Maintenance: consider migration from karma-runner to jasmine-browser-runner.
### Describe the bug

The Karma browser-based test framework that we use to run tests for the Sphinx JavaScript code that runs in the client browser is [deprecated](https://github.com/karma-runner/karma/commit/450fdfdac5b999967daec1020f1ac69cf9b854ab).

We may want to consider migration.  The `karma` project's deprecation notice suggests [`jasmine-browser-runner`](https://github.com/jasmine/jasmine-browser-runner) as an alternative.

### How to Reproduce

N/A

### Environment Information

```text
N/A
```

### Sphinx extensions

```python
N/A
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jayaddison**
### Feature or Bugfix
- Refactoring

### Purpose
- Migrate from the deprecated `karma` JavaScript test framework to maintained `jasmine` equivalents.

### Detail
- Update `package.json` dependencies -- remove `karma`, add `jasmine` dependencies.
- Use `npx jasmine-browser-runner init` to initialize the project, and migrate relevant `karma.conf.js` settings.
- Update the GitHub Actions workflows accordingly.
- No modifications to unit tests, fixtures, or application JavaScript code.

### Relates
- Resolves #12141.

### CODE REVIEW

File: spec/support/jasmine-browser.mjs
Comment (AA-Turner) [inline spec/support/jasmine-browser.mjs:1]: Is this location configurable?

Comment (jayaddison) [reply spec/support/jasmine-browser.mjs:1]: It is, yep - we can use an arbitrary path, and then provide a `--config` argument to the `jasmine-browser-runner` command.

Comment (AA-Turner) [reply spec/support/jasmine-browser.mjs:1]: perhaps somewhere sensible in tests/js?

**Comment by jayaddison** [comment]
Thanks @AA-Turner!

**Pull request merged**